### ISSUE TITLE
PI0Policy.from_pretrained returns an untrained model when loading fails, even with strict=True

### ISSUE THREAD

**Issue by shoumikhin** [huggingface/lerobot#4577]
### What happens

When `from_pretrained` cannot apply a checkpoint to the model, it prints a warning and then returns the model anyway. The model that comes back still has its freshly initialized weights, so it is untrained.

The call looks successful. Nothing raises, and the return value is a normal policy object.

### Why this is a problem

Anything that compares the loaded model against itself will pass. In my case I was checking that an exported model matched the original one. Both sides were the same untrained model, so the numbers agreed perfectly and told me nothing. A whole batch of measurements was worthless before I noticed, and the only clue was one line of printed text scrolled far up in a log.

A load that fails is much cheaper to deal with when it stops immediately.

### The `strict` argument does not help

`from_pretrained` accepts `strict`, and it defaults to `True`. That value does reach `load_state_dict`, which raises as expected. But the surrounding `try` block catches that same exception, prints it, and falls through to `return model`.

So a caller who asks for strict loading, or who simply relies on the default, still gets an untrained model back and no exception.

The base class behaves the way I expected: `_load_as_safetensor` in `policies/pretrained.py` passes `strict` down and lets the error surface.

### How to reproduce

Load a pi0.5 checkpoint with the pi0 policy class. This is an easy mistake to make, because both live under a `pi0` name and the checkpoint is accepted without complaint:

```python
import torch
from lerobot.policies.pi0.modeling_pi0 import PI0Policy

policy = PI0Policy.from_pretrained("lerobot/pi05_base", strict=True)

zeros = [
    name
    for name, t in policy.state_dict().items()
    if torch.is_floating_point(t) and t.numel() and float(t.detach().abs().max()) == 0.0
]
print(type(policy).__name__, "returned with", len(zeros), "all-zero tensors")
```

Output:

```
Warning: Could not load state dict: Error(s) in loading state_dict for PI0Policy:
        Unexpected key(s) in state_dict: "...gemma_expert.model.layers.0.input_layernorm.dense.weight", ...
PI0Policy returned with 37 all-zero tensors
```

The 37 tensors are layer norms that never received their weights. The model runs and produces output, it is just not the trained network.

To be clear about my own mistake: using `PI0Policy` here is wrong, and `PI05Policy` in `policies/pi05/` loads this checkpoint correctly. The report is not about the class mix-up. It is that the wrong class produced a usable-looking object instead of an error.

The same shape happens for other reasons too. On version 0.6.1, `PI0Policy.from_pretrained("lerobot/pi0_base")` hit a key mismatch and returned a fully random model, again with only a printed warning. That mismatch is fixed in 0.6.2, but the silent return is not.

### Where the code is

Three policies define their own `from_pretrained` with this pattern:

- `policies/pi0/modeling_pi0.py`, line 895
- `policies/pi05/modeling_pi05.py`, line 1041
- `policies/pi0_fast/modeling_pi0_fast.py`, line 907

There is a second one a little earlier in each file, for the case where the checkpoint file cannot be fetched at all. That one is clearer, because it says "Returning model without loading pretrained weights" out loud, but it still returns an unusable model to a caller who asked for a pretrained one.

### Suggestion

Let the error propagate when `strict` is true, since that is what the argument is for. If there is a reason to keep going for some callers, `strict=False` is the natural way to ask for it.

If raising would break existing users, then a way to ask the returned object whether the weights were applied would at least make the failure detectable in code rather than only in printed output.

I am happy to send a pull request if you tell me which of the two you would prefer.

### Version

`0.6.2`, Python 3.12, Linux aarch64, `transformers` 5.5.4.

**Comment by Shaik-Yaseen1** [comment]
Hey hi this yaseen coming here to make contribution for this repo here i will face lot of things for setup locally can u help me to setup this locally, i really very curious to understand and working on the robotics

<img width="964" height="1016" alt="Image" src="https://github.com/user-attachments/assets/7a9f754c-8d89-4295-853e-d43808220b6b" />

**Comment by sattyamjjain** [comment]
Filed one day ago, and it deserves to be treated as an evaluation-validity bug rather than a loading bug, because of what it does downstream.

A silently untrained policy does not crash. It produces well-formed action chunks at the right shape and cadence, which means every metric a harness computes on it looks like a real number. Your export-versus-original comparison agreed perfectly, and the reason it agreed is the worst possible one. Anyone running a batch sweep would have shipped that agreement as a result.

Two suggestions, one for the fix and one for everyone downstream of it.

For the fix: `strict=True` should propagate, not warn. A warning printed inside a loop of 200 evaluations is invisible.

For downstream: a cheap canary is to assert the loaded policy's output is not distributionally identical to a freshly initialised one on a fixed batch of inputs. That catches this failure and also catches a partially loaded checkpoint, which warns today and is equally silent in aggregate.
